libnavigation-base/src/main/java/com/mapbox/navigation/base/utils/route/RouteUtils.kt

```diff
@@ -0,0 +1,163 @@
+
+            routeLeg.closures()?.forEach { closure ->
+                val silentWaypointsInClosureRange = silentWaypoints.inGeometryRange(
+                    closure.geometryIndexStart(), closure.geometryIndexEnd()
+                )
+
+                if (closure.geometryIndexStart() == 0 &&
+                    snappingResultList.getOrNull(index) != true
+                ) {
+                    logW(
+                        "Route with id [${navigationRoute.id}] has closure at the start of the" +
+                            " leg, leg index $routeLegIndex",
+                        LOG_CATEGORY
+                    )
+                    return true
+                }
+                if (closure.geometryIndexEnd() == legLastGeometryIndex &&
+                    snappingResultList.getOrNull(index + silentWaypoints.size + 1) != true
+                ) {
+                    logW(
+                        "Route with id [${navigationRoute.id}] has closure at the end of the" +
+                            " leg, leg index $routeLegIndex",
+                        LOG_CATEGORY
+                    )
+                    return true
+                }
+                if (silentWaypointsInClosureRange.isNotEmpty()) {
+                    silentWaypointsInClosureRange.forEach {
+                        val isSnapAllowed =
+                            snappingResultList.getOrNull(index + 1 + it) ?: false
+                        if (!isSnapAllowed) {
+                            logW(
+                                "Route with id [${navigationRoute.id}] has closure at leg " +
+                                    "index $routeLegIndex, that overlaps silent (via) waypoint",
+                                LOG_CATEGORY
+                            )
+                            return true
+                        }
+                    }
+                }
+                if (closure.geometryIndexStart() != 0 &&
+                    closure.geometryIndexEnd() != legLastGeometryIndex &&
+                    silentWaypointsInClosureRange.isEmpty()
+                ) {
+                    logW(
+                        "Route with id [${navigationRoute.id}] has closure at leg " +
+                            "index $routeLegIndex",
+                    )
+                    return true
+                }
+            }
+            index += silentWaypoints.size
+        }
+
+        return false
+    }
+
+    private fun DirectionsRoute.getSnappingResultList(): List<Boolean> {
+        val snappingIncludeClosuresList = routeOptions()?.snappingIncludeClosuresList()
+        val snappingIncludeStaticClosuresList = routeOptions()?.snappingIncludeStaticClosuresList()
+
+        val snappingResultList = mutableListOf<Boolean>()
+        for (
+            index in 0 until maxOf(
+                snappingIncludeClosuresList?.size ?: 0,
+                snappingIncludeStaticClosuresList?.size ?: 0
```

Comment:
just curious, is it possible for them to have a different size?

Reply:
No, a request would fail.

However, one can be `null` and the other filled up.